dask/dataframe/tests/test_io.py

```diff
@@ -697,6 +697,61 @@ def test_to_hdf():
     a = dd.from_pandas(df, 1)
     with tmpfile('h5') as fn:
         a.to_hdf(fn, '/data')
+        out = pd.read_hdf(fn, '/data')
+        tm.assert_frame_equal(df, out[:])
+
+    # saving to multiple datasets
+    a = dd.from_pandas(df, 2)
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data*')
+        out = dd.read_hdf(fn, '/data*')
+        tm.assert_frame_equal(df, out.compute())
+
+    # saving to multiple files
+    a = dd.from_pandas(df, 2)
+    with tmpdir() as dn:
+        fn = os.path.join(dn, 'data_*.h5')
+        a.to_hdf(fn, '/data')
+        out = dd.read_hdf(fn, '/data')
+        tm.assert_frame_equal(df, out.compute())
+
+    # saving to multiple datasets with custom name_function
+    a = dd.from_pandas(df, 2)
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data_*', name_function=lambda i: 'a' * (i +  1))
+        out = dd.read_hdf(fn, '/data_*')
+        tm.assert_frame_equal(df, out.compute())
+
+        out = pd.read_hdf(fn, '/data_a')
+        tm.assert_frame_equal(out, df.iloc[:2])
+        out = pd.read_hdf(fn, '/data_aa')
+        tm.assert_frame_equal(out, df.iloc[2:])
+
+    # saving to multiple files with custom name_function
+    a = dd.from_pandas(df, 2)
+    with tmpdir() as dn:
+        fn = os.path.join(dn, 'data_*.h5')
+        a.to_hdf(fn, '/data', name_function=lambda i: 'a' * (i +  1))
+        out = dd.read_hdf(fn, '/data')
+        tm.assert_frame_equal(df, out.compute())
+
+        out = pd.read_hdf(os.path.join(dn, 'data_a.h5'), '/data')
+        tm.assert_frame_equal(out, df.iloc[:2])
+        out = pd.read_hdf(os.path.join(dn, 'data_aa.h5'), '/data')
+        tm.assert_frame_equal(out, df.iloc[2:])
+
+    # saving to different datasets in multiple files with custom name_function
+    a = dd.from_pandas(df, 2)
+    with tmpdir() as dn:
+        fn = os.path.join(dn, 'data_*.h5')
+        a.to_hdf(fn, '/data_*', name_function=lambda i: 'a' * (i +  1))
+        out = dd.read_hdf(fn, '/data_*')
+        tm.assert_frame_equal(df, out.compute())
+
+        out = pd.read_hdf(os.path.join(dn, 'data_a.h5'), '/data_a')
+        tm.assert_frame_equal(out, df.iloc[:2])
+        out = pd.read_hdf(os.path.join(dn, 'data_aa.h5'), '/data_aa')
+        tm.assert_frame_equal(out, df.iloc[2:])
```

Comment:
This last test feels a bit odd.  It's not clear to me what proper behavior is when both the path and the datapath have a `*` in them.  I think it's unlikely that the user wants the same naming scheme used in both.  I would probably opt to err in this case.